bob-apps/bin/bobc.rs

```diff
@@ -1,137 +1,421 @@
+                (FilePattern::WithRE(re, _), Some(key_pattern)) => {
+                    Box::new(re.get_filenames(&key_pattern).map(|(key, name)| KeyName {key,name}))
+                }
+                (FilePattern::WithoutRE(path), Some(KeyPattern::Single(key))) => {
+                    Box::new(iter::once(KeyName {
+                        key,
+                        name: path.to_string()
+                    }))
+                }
+                (FilePattern::WithoutRE(path), Some(KeyPattern::Range(r))) => {
+                    Box::new(r.map(move |key| KeyName {
+                        key,
+                        name: path.to_string()
+                    }))
+                }
+                (FilePattern::WithoutRE(path), Some(KeyPattern::Multiple(v))) => {
+                    Box::new(v.into_iter().map(move |key| KeyName {
+                        key,
+                        name: path.to_string()
+                    }))
+                }
+                (FilePattern::WithoutRE(_), None) => {
+                    return error!("Key arg is required if not using file pattern");
+                }
+            };
+            for kn in keys_names {
+                put(kn.key, app_args.keysize, &kn.name, &mut client).await;
             }
-            "get" => {
-                let addr = sub_mathes
-                    .value_of("uri")
-                    .expect("has default value")
-                    .parse()
-                    .expect("wrong format of url");
-                info!("GET key:\"{:?}\" from  \"{}\"", key, addr);
-                get(key, addr).await;
+        }
+        GET_SC => {
+            let keys_names = match (app_args.file_pattern.unwrap(), app_args.key_pattern.unwrap()) {
+                (FilePattern::WithRE(re, _), key) => {
+                    let files = re.get_filenames(&key);
+                    prepare_get(files)
+                }
+                (FilePattern::WithoutRE(path), KeyPattern::Single(key)) => {
+                    prepare_get(Box::new(iter::once((key, path.to_string()))))
+                }
+                (FilePattern::WithoutRE(_), _) => {
+                    return error!("Multiple keys are not allowed without pattern")
+                }
+            };
+            for kn in keys_names {
+                get(kn.key, app_args.keysize, &kn.name, &mut client).await;
             }
-            _ => {}
         }
+        EXISTS_SC => {
+            match app_args.key_pattern.unwrap() {
+                KeyPattern::Single(val) => {
+                    exist(vec![val], app_args.keysize, &mut client).await;
+                }
+                KeyPattern::Range(range) => {
+                    exist(range.collect(), app_args.keysize, &mut client).await;
+                }
+                KeyPattern::Multiple(vec) => {
+                    exist(vec, app_args.keysize, &mut client).await;
+                }
+            }
+        }
+        _ => unreachable!(),
```

Comment:
This should not be `unreachable`. It is better to `panic` here with message: `unknown command {}`

Reply:
Fix: [34405fe](https://github.com/qoollo/bob/pull/562/commits/34405feb23d5693e1db1b429d46c66fdb930d19c)